openupgradelib/openupgrade.py

```diff
@@ -887,24 +887,65 @@ def rename_models(cr, model_spec):
     # TODO: signal where the model occurs in references to ir_model
 
 
-def rename_xmlids(cr, xmlids_spec):
+def rename_xmlids(cr, xmlids_spec, allow_merge=False):
     """
     Rename XML IDs. Typically called in the pre script.
     One usage example is when an ID changes module. In OpenERP 6 for example,
     a number of res_groups IDs moved to module base from other modules (
     although they were still being defined in their respective module).
 
     :param xmlids_spec: a list of tuples (old module.xmlid, new module.xmlid).
+    :param allow_merge: if the new ID already exists, try to merge the records.
+    This is recommended when renaming module categories, which are generated
+    on the fly by the Odoo database initialization routine and may resurface
+    over a longer period of time.
```

Comment:
to greenify ci

```yml
Warning, treated as error:

/home/travis/build/OCA/openupgradelib/openupgradelib/openupgrade.py:docstring of openupgradelib.openupgrade.rename_xmlids:8:Field list ends without a blank line; unexpected unindent.
```

not sure if it's this:
```suggestion
    :param allow_merge: if the new ID already exists, try to merge the records.
     This is recommended when renaming module categories, which are generated
     on the fly by the Odoo database initialization routine and may resurface
     over a longer period of time.
```

or this:
```suggestion

    :param allow_merge: if the new ID already exists, try to merge the records.
    This is recommended when renaming module categories, which are generated
    on the fly by the Odoo database initialization routine and may resurface
    over a longer period of time.
```

or both

Reply:
Thanks for suggesting this fix!